nautobot/extras/migrations/0072_migrate_custom_field_data.py

```diff
@@ -0,0 +1,62 @@
+# Generated by Django 3.2.18 on 2023-03-17 17:25
+
+from django.db import migrations
+from nautobot.core.models.fields import slugify_dashes_to_underscores
+
+
+def generate_unique_custom_field_slug_and_migrate_custom_field_data(apps, schema_editor):
+    CustomField = apps.get_model("extras", "customfield")
+    ContentType = apps.get_model("contenttypes", "ContentType")
+
+    # Make sure that the key is unique during migration
+    cf_keys = []
```

Comment:
This is kinda like a record for `cf.key` that we have already encountered in the for loop. So it should be empty at first so that as we go through the `CustomField` instances if we have duplicate `keys` we append indices at the end of the `key`. If we fill it out first, as we go through the instances, every `key` would be changed.

Reply:
I guess maybe I was thinking of something like:

1. key = "Custom Field"
2. key = "custom-field"
3. key = "custom_field_2"

On processing, as currently implemented, 1 becomes `custom_field`, 2 becomes `custom_field_2`, and 3 gets renamed to `custom_field_2_2`, which doesn't seem like the best experience? Admittedly this is a contrived corner case.

Reply:
I see. The corner case definitely is something to consider. 
But we do have the `print` statement (mapping the old_key to the new_key) so that the user can easily find the `CustomField` with the weird new `key` and modify them by hand. and the corner case might not be as prevalent as we think.

What do you think? I am definitely open to suggestions on how to make this processing algorithm more robust.

Reply:
Remember that `key` is not user editable for existing custom fields, but this is probably good enough for now.